src/GrumPHP/Task/PhpVersion.php

```diff
@@ -0,0 +1,114 @@
+<?php
+
+namespace GrumPHP\Task;
+
+use GrumPHP\Configuration\GrumPHP;
+use GrumPHP\Runner\TaskResult;
+use GrumPHP\Runner\TaskResultInterface;
+use GrumPHP\Task\Context\ContextInterface;
+use GrumPHP\Task\Context\RunContext;
+use Symfony\Component\OptionsResolver\OptionsResolver;
+
+/**
+ * PhpVersion task
+ */
+class PhpVersion implements TaskInterface
+{
+    /**
+     * @var \GrumPHP\Util\PhpVersion
+     */
+    private $phpVersionUtility;
+
+    /**
+     * @var GrumPHP
+     */
+    private $grumPHP;
+
+    /**
+     * PhpVersion constructor.
+     * @param GrumPHP $grumPHP
+     * @param \GrumPHP\Util\PhpVersion $phpVersionUtility
+     */
+    public function __construct(GrumPHP $grumPHP, \GrumPHP\Util\PhpVersion $phpVersionUtility)
+    {
+        $this->grumPHP = $grumPHP;
+        $this->phpVersionUtility = $phpVersionUtility;
+    }
+
+    /**
+     * This methods specifies if a task can run in a specific context.
+     *
+     * @param ContextInterface $context
+     *
+     * @return bool
+     */
+    public function canRunInContext(ContextInterface $context)
+    {
+        return $context instanceof RunContext;
```

Comment:
Can you also add the PreCommitContext? This way it is checked both during commit and in run modus.